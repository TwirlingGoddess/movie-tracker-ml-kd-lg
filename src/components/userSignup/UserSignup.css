.submit-form {
  height: 8rem;
  display: flex;
  flex-direction: column;
  margin-top: .8rem;
  justify-content: space-around;
  align-items: center; }

.signup-title {
  font-size: 1rem;
  margin: 0; }

.signup-input {
  width: 14rem;
  height: 1rem; }

.user-submit-button {
  width: 6rem; }
