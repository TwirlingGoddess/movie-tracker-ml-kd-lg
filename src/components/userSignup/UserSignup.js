import React, { Component } from 'react'
import { newUserFetchCall } from '../../helpers.js'


class UserSignup extends Component {
  constructor() {
    super()
    this.state = {
      name: '',
      email: '',
      password: ''
    }
  }

  handleSubmit = async event => {
    event.preventDefault()
    const { name, email, password } = this.state
    const data = await newUserFetchCall(name, email,password)
    console.log(data)
  }
  
  handleChange = (event) => {
    const { name, value } = event.target
    this.setState({
      [name]: value
    })
  } 

  render() {
    return (
      <div>
        <form onSubmit={this.handleSubmit}>
          <h2>Sign Up</h2>
          <input placeholder='User Name' value={this.state.name} type='text' name='name' onChange={this.handleChange}/>
          <input placeholder='Email' value={this.state.email} type='email' name='email' onChange={this.handleChange}/>
          <input placeholder='Password' value={this.state.password} type='password' name='password' onChange={this.handleChange}/>
          <button>Sign Up</button>
        </form>
        <button>Cancel</button>
      </div>
    )
  }
}

export default UserSignup