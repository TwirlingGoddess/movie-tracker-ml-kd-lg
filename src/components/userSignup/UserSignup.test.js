import React from 'react'
import UserSignup from './UserSignup'
import { shallow } from 'enzyme'

describe('UserSignup', () => {
  let wrapper
  let mockHandleSignup = jest.fn()
  let mockState = {name: '', email: '', password: ''}

  beforeEach(() => {
    wrapper = shallow(<UserSignup handleSignup={mockHandleSignup} />)
  })

  
  it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
  })

  it('should reset the input fields after user submits info', async () => {
    wrapper.setState ({name: 'kiel', email: 'kiel@gmail.com', password: 'password'})
    const mockEvent = new Event('event') 
    
    await wrapper.instance().handleSubmit(mockEvent)

    expect(wrapper.state('name')).toEqual('')
    expect(wrapper.state('email')).toEqual('')
    expect(wrapper.state('password')).toEqual('')
  })

  it('should call handleSignup when the form is submitted', () => {
    wrapper.find('#user-submit-button').simulate('click')

    expect(mockHandleSignup).toHaveBeenCalled()
  })

  it('should call handleSubmit when the form is submitted', async () => {
    const mockEvent = new Event('event') 

    await wrapper.instance().handleSubmit(mockEvent)

    expect(mockHandleSignup).toHaveBeenCalled()
  })

  it('should set the state with the names and values', () => {
    const mockData = {target: { name: 'name', value: 'Kiel' }}

    wrapper.instance().handleChange(mockData)

    expect(wrapper.state('name')).toEqual('Kiel')
  }) 

  it('should enable the button when there is something in all of the inputs', () => {
    let result = wrapper.instance().validateButton()

    expect(result).toEqual(false)

    wrapper.setState({name: 'kiel', email: 'kiel@kiel', password: 'password'})
    result = wrapper.instance().validateButton()

    expect(result).toEqual(true)
  })
})