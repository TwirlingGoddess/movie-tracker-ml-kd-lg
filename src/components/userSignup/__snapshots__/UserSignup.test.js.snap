// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSignup should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserSignup
    handleSignup={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="submit-form"
        onSubmit={[Function]}
>
        <h2
                className="signup-title"
        >
                Sign Up
        </h2>
        <input
                className="signup-input"
                name="name"
                onChange={[Function]}
                placeholder="User Name"
                type="text"
                value=""
        />
        <input
                className="signup-input"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="email"
                value=""
        />
        <input
                className="signup-input"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
        />
        <div>
                <button
                        className="user-submit-button"
                >
                        Sign Up
                </button>
                <button
                        className="user-submit-button"
                        id="user-submit-button"
                        onClick={[Function]}
                >
                        Cancel
                </button>
        </div>
</form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="signup-title"
>
            Sign Up
</h2>,
          <input
            className="signup-input"
            name="name"
            onChange={[Function]}
            placeholder="User Name"
            type="text"
            value=""
/>,
          <input
            className="signup-input"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="email"
            value=""
/>,
          <input
            className="signup-input"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
/>,
          <div>
            <button
                        className="user-submit-button"
            >
                        Sign Up
            </button>
            <button
                        className="user-submit-button"
                        id="user-submit-button"
                        onClick={[Function]}
            >
                        Cancel
            </button>
</div>,
        ],
        "className": "submit-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
            "className": "signup-title",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "signup-input",
            "name": "name",
            "onChange": [Function],
            "placeholder": "User Name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "signup-input",
            "name": "email",
            "onChange": [Function],
            "placeholder": "Email",
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "signup-input",
            "name": "password",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="user-submit-button"
>
                Sign Up
</button>,
              <button
                className="user-submit-button"
                id="user-submit-button"
                onClick={[Function]}
>
                Cancel
</button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Up",
                "className": "user-submit-button",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Cancel",
                "className": "user-submit-button",
                "id": "user-submit-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Cancel",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="submit-form"
          onSubmit={[Function]}
>
          <h2
                    className="signup-title"
          >
                    Sign Up
          </h2>
          <input
                    className="signup-input"
                    name="name"
                    onChange={[Function]}
                    placeholder="User Name"
                    type="text"
                    value=""
          />
          <input
                    className="signup-input"
                    name="email"
                    onChange={[Function]}
                    placeholder="Email"
                    type="email"
                    value=""
          />
          <input
                    className="signup-input"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
          />
          <div>
                    <button
                              className="user-submit-button"
                    >
                              Sign Up
                    </button>
                    <button
                              className="user-submit-button"
                              id="user-submit-button"
                              onClick={[Function]}
                    >
                              Cancel
                    </button>
          </div>
</form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="signup-title"
>
              Sign Up
</h2>,
            <input
              className="signup-input"
              name="name"
              onChange={[Function]}
              placeholder="User Name"
              type="text"
              value=""
/>,
            <input
              className="signup-input"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="email"
              value=""
/>,
            <input
              className="signup-input"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
/>,
            <div>
              <button
                            className="user-submit-button"
              >
                            Sign Up
              </button>
              <button
                            className="user-submit-button"
                            id="user-submit-button"
                            onClick={[Function]}
              >
                            Cancel
              </button>
</div>,
          ],
          "className": "submit-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
              "className": "signup-title",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "signup-input",
              "name": "name",
              "onChange": [Function],
              "placeholder": "User Name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "signup-input",
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email",
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "signup-input",
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="user-submit-button"
>
                  Sign Up
</button>,
                <button
                  className="user-submit-button"
                  id="user-submit-button"
                  onClick={[Function]}
>
                  Cancel
</button>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign Up",
                  "className": "user-submit-button",
                },
                "ref": null,
                "rendered": "Sign Up",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Cancel",
                  "className": "user-submit-button",
                  "id": "user-submit-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Cancel",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
